application/artifacts/artifacts_view.py

```diff
@@ -69,21 +70,70 @@ def _find_multiple_by(params):
     return artifacts.order_by('created_at')[_from:_to]
 
 
-class ArtifactsController(ApplicationController):
-    """ Controller for Artifacts """
-
-    method_decorators = [check_es_connection]
+class ArtifactView(MethodResource):
+    """ Access Artifacts by id """
 
-    def show(self, object_id):
+    @use_kwargs(artifacts_validator.get_args())
+    @marshal_with(ARTIFACT_SCHEMA)
+    def get(self, **params):
         """Logic for getting a single artifact"""
         try:
-            artifact = Artifact.find_by(id_=object_id)
-            return respond_with(artifact)
+            artifact = Artifact.find_by(id_=params['id'])
+            return artifact
+        except Artifact.DoesNotExist:
+            return abort(404, 'artifact not found')
+
+    @use_kwargs(artifacts_validator.update_args())
+    @marshal_with(None, code=204)
+    def put(self, **params):
+        """Logic for updating an artifact"""
+        try:
+            artifact = Artifact.find_by(id_=params.pop("id"))
+            builder = ArtifactBuilder.for_artifact(artifact)
+            builder.update_with(**params)
+            return no_content()
         except Artifact.DoesNotExist:
-            return {"error": "not found"}, 404
+            return abort(404, 'artifact not found')
+
+    @use_kwargs(artifacts_validator.update_args())
+    @marshal_with(None, code=204)
+    def patch(self, **params):
+        """Logic for updating an artifact"""
+        try:
+            artifact = Artifact.find_by(id_=params.pop("id"))
+            builder = ArtifactBuilder.for_artifact(artifact)
+            builder.update_with(**params)
+            return no_content()
+        except Artifact.DoesNotExist:
+            return abort(404, 'artifact not found')
+
```

Comment:
I'm not the biggest fan of copy pasting everything for put and patch when it's the same.
I would either move the method into update and call that from both (you could even change the name to update and 
define two lambdas for PUT and PATCH that pass the call to update) or override the MethodView class to call PUT when PATCH isn't defined and vice versa.

Reply:
Patch and put should probably not do the same thing per default. I like having that explicitly. I will call the other method from the function though. So we don't have actual code duplication.

That way you can see that for example: PUT does actually just call Patch and when you want to have an actual PUT you'll have to reimplement it. Thoughts?

Reply:
You can check users, teams and artifacts controller. Sometimes PATCH is actually just a PUT and sometimes it's vice versa. So I am calling the respective one.